SU2_CFD/src/solver_direct_elasticity.cpp

```diff
@@ -225,22 +225,26 @@ CFEASolver::CFEASolver(CGeometry *geometry, CConfig *config) : CSolver() {
   else nSolVar = nVar;
   
   SolRest = new su2double[nSolVar];
+  for (iVar = 0; iVar < nSolVar; iVar++) SolRest[iVar] = 0.0;
 
-  /*--- Initialize from zero everywhere. ---*/
-  long iVertex;
-  bool isVertex;
+  /*--- Initialize from zero everywhere ---*/
 
-  for (iVar = 0; iVar < nSolVar; iVar++) SolRest[iVar] = 0.0;
-  for (iPoint = 0; iPoint < nPoint; iPoint++) {
-    isVertex = false;
+  node = new CFEABoundVariable(SolRest, nPoint, nDim, nVar, config);
+  snode = static_cast<CFEABoundVariable*>(node);
+
+  /*--- Set which points are vertices and allocate boundary data. ---*/
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++)
     for (unsigned short iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
-      iVertex = geometry->node[iPoint]->GetVertex(iMarker);
-      if (iVertex != -1){isVertex = true; break;}
+      long iVertex = geometry->node[iPoint]->GetVertex(iMarker);
+      if (iVertex >= 0) {
+        snode->Set_isVertex(iPoint,true);
+        break;
+      }
     }
-    if (isVertex) node[iPoint] = new CFEABoundVariable(SolRest, nDim, nVar, config);
-    else          node[iPoint] = new CFEAVariable(SolRest, nDim, nVar, config);
-  }
-  
+  snode->AllocateBoundaryVariables(config);
```

Comment:
@rsanfer this should address your concerns about memory wastage, storage for surface forces is allocated after the "isVertex" booleans are set. If the code is compiled without NDEBUG checks are performed to make sure the data is not being accessed before a call to Allocate, or after new calls to Set_isVertex.
Would this strategy work for the new mesh solver?

Reply:
I've also changed the CFEABoundVariable and split it into 2 classes depending on whether there are Fluid loads or not. We can take a look. I plan to open the PR for the mesh solver hopefully today or at latest on Monday, there we can discuss how we could combine the developments.